File: client/src/app/models/userParams.ts
import { User } from "./user.model";

export class UserParams {   
    gender: string;
    minAge: number;
    maxAge: number;
    pageNumber: number;
    pageSize: number;
    city: string;

    constructor(user: User) {
        this.gender = user.gender;
        this.pageNumber = 1;
        this.pageSize = 5;
        this.minAge = 1;
        this.maxAge = 15;
        this.city = 'Москва';
    }
}

File: API/Helpers/UserParams.cs
namespace API.Helpers
{
    public class UserParams : PaginationParams
    {
        public string CurrentUsername { get; set; }
        public string Gender { get; set; }
        public int MinAge { get; set; } = 1;
        public int MaxAge { get; set; } = 150;
        public string OrderBy { get; set; } = "lastActive";
    }
}